### PR TITLE
Moves VM details to Job model allowing differing images/flavors/etc

### LINKED ISSUE #115: Supply VM details via API with jobs
For a job to specify its VM image and other details, these things must be provided by the API. Currently this is hard-coded in the lando server config and should be moved to the API layer.

Dependent of https://github.com/Duke-GCB/lando/issues/92

### PR DESCRIPTION
**Pull request by dleehr**
- Adds VMSettings and CloudSettings models, populated via migrations
- Refactors Job and JobQuestionnaire to reference these settings so that a Job object contains all of the `vm_settings` from the lando server (image name, openstack project, volume mounts, cwl command)
- Refactors JobFactory to incorporate these new settings, copying from the questionnaire to the job
- Updates `loadworkflow` command to expect the name of a VMSettings instead of a VM project name
- Adds nested fields to AdminJobSerializer, allowing admin clients to get VM and cloud settings in the job payload 

Tested migrations by loading current dev and prod databases into local postgres then migrating.

The migrations will only fill out the bare minimum fields for VMSettings and CloudSettings. These must be further populated from what's currently in the lando server configs, and should be addressed at deploy time by a fixture.

Some issues in the migrations were not discovered until trying to migrate on postgres. Created  #119.

Fixes #115

### CODE REVIEW

**Comment by dleehr** [comment]
I'll be testing this out on dev today, so more commits may come.

File: data/models.py [hunk 1]
```diff
@@ -117,6 +117,61 @@ def __unicode__(self):
...
+    ssh_key_name = models.CharField(max_length=255, help_text='Name of SSH key to inject into VM on launch')
+    network_name = models.CharField(max_length=255, help_text='Name of network to attach VM to on launch')
+    allocate_floating_ips = models.BooleanField(blank=False, default=False,
+                                                help_text='Allocate floating IPs to launched VMs')
+    floating_ip_pool_name = models.CharField(max_length=255, blank=True, null=True,
+                                             help_text='Name of floating IP pool to allocate from')
+
+    def __unicode__(self):
+        return '{}: id {}, Proj: {}'.format(self.name, self.pk, self.vm_project.name)
+
```
Comment (johnbradley) [inline data/models.py:153]: For consistency with VMSettings do you want to add something like:
```
class Meta:
    verbose_name_plural = "Cloud Settings Collections"
```

Right now the admin interface shows this as `Cloud settingss`.

File: data/models.py [hunk 2]
```diff
@@ -280,22 +317,25 @@ class JobQuestionnaire(models.Model):
...
-    share_group = models.ForeignKey(ShareGroup, blank=False, null=False,
-                                    help_text='Users who will have job output shared with them')
+    volume_mounts = models.TextField(null=False, blank=False, default=json.dumps({'/dev/vdb1': '/work'}),
+                                     help_text='JSON-encoded dictionary of volume mounts, e.g. {"/dev/vdb1": "/work"}')
 
     def __unicode__(self):
         return '{} desc:{}'.format(self.id, self.description)
 
 
+
```
Comment (johnbradley) [inline data/models.py]: Extra blank line (PEP8).

File: data/models.py [hunk 3]
```diff
@@ -169,14 +224,14 @@ class Job(models.Model):
     step = models.CharField(max_length=1, choices=JOB_STEPS, null=True, blank=True,
                             help_text="Job step (progress within Running state)")
     last_updated = models.DateTimeField(auto_now=True, blank=False)
-    vm_flavor = models.CharField(max_length=255, blank=False, default='m1.small',
-                                 help_text="Determines CPUs and RAM VM allocation used to run this job.")
+    vm_settings = models.ForeignKey(VMSettings, null=False, blank=False,
+                                    help_text='Collection of settings to use when launching VM for this job')
+    vm_flavor = models.ForeignKey(VMFlavor, null=False,
+                                  help_text='VM Flavor to use when creating VM instances for this questionnaire')
```
Comment (johnbradley) [inline data/models.py]: Shouldn't this help_text end with `for this job` instead of `for this questionnaire`?

**Review by johnbradley — CHANGES_REQUESTED** [review]
Requesting some minor tweaks.
Overall LGTM.

**Review by johnbradley — APPROVED** [review]

**Pull request merged**